lxd/container.go

```diff
@@ -660,6 +681,37 @@ func containerCreateInternal(d *Daemon, args containerArgs) (container, error) {
 		}
 	}
 
+	// Check that there are no contradicting root disk devices.
+	var profileRootDiskDevices []string
+	for _, pName := range args.Profiles {
+		_, p, err := dbProfileGet(d.db, pName)
+		if err != nil {
+			return nil, fmt.Errorf("Could not load profile '%s'.", pName)
+		}
+
+		k, v := containerGetRootDiskDevice(p.Devices)
+		if k != "" && v["pool"] == "" {
+			return nil, fmt.Errorf("A root disk device must have the \"pool\" property set.")
+		} else if k != "" && !shared.StringInSlice(k, profileRootDiskDevices) {
+			profileRootDiskDevices = append(profileRootDiskDevices, k)
+		}
+	}
+
+	k, newLocalRootDiskDevice := containerGetRootDiskDevice(args.Devices)
+	// Check whether container has a local root device with a "pool"
+	// property set.
+	if k != "" && newLocalRootDiskDevice["pool"] == "" {
+		return nil, fmt.Errorf("A root disk device must have the \"pool\" property set.")
```

Comment:
No, it shouldn't because before containerCreateInternal() is called, createFromMigration() will set a storage pool. In my testing it worked. But I'd prefer if you did some tests yourself too. :)